native/src/main/java/io/ballerina/observe/metrics/newrelic/NewRelicMetricsReporter.java

```diff
@@ -0,0 +1,189 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.ballerina.observe.metrics.newrelic;
+
+import com.newrelic.telemetry.Attributes;
+import com.newrelic.telemetry.OkHttpPoster;
+import com.newrelic.telemetry.TelemetryClient;
+import com.newrelic.telemetry.metrics.Count;
+import com.newrelic.telemetry.metrics.MetricBatch;
+import com.newrelic.telemetry.metrics.MetricBuffer;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.observability.metrics.Counter;
+import io.ballerina.runtime.observability.metrics.DefaultMetricRegistry;
+import io.ballerina.runtime.observability.metrics.Gauge;
+import io.ballerina.runtime.observability.metrics.Metric;
+import io.ballerina.runtime.observability.metrics.MetricConstants;
+import io.ballerina.runtime.observability.metrics.MetricId;
+import io.ballerina.runtime.observability.metrics.PercentileValue;
+import io.ballerina.runtime.observability.metrics.PolledGauge;
+import io.ballerina.runtime.observability.metrics.Snapshot;
+import io.ballerina.runtime.observability.metrics.Tag;
+
+import java.io.PrintStream;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static io.ballerina.observe.metrics.newrelic.ObserveNativeImplConstants.EXPIRY_TAG;
+import static io.ballerina.observe.metrics.newrelic.ObserveNativeImplConstants.PERCENTILE_TAG;
+
+/**
+ * This is the New Relic metric reporter class.
+ */
+public class NewRelicMetricsReporter {
+    private static final PrintStream console = System.out;
+    private static final String METRIC_REPORTER_ENDPOINT = "https://metric-api.newrelic.com/metric/v1";
+    private static final int SCHEDULE_EXECUTOR_INITIAL_DELAY = 0;
+
+    public static void sendMetrics(BString apiKey, int metricReporterFlushInterval,
+                                         int metricReporterClientTimeout) {
+        // create a TelemetryClient with an HTTP connect timeout of 10 seconds.
+        TelemetryClient telemetryClient =
+                TelemetryClient.create(
+                        () -> new OkHttpPoster(Duration.of(metricReporterClientTimeout, ChronoUnit.MILLIS)),
+                        apiKey.getValue());
+        Attributes commonAttributes = null;
+        try {
+            commonAttributes = new Attributes()
+                    .put("host", InetAddress.getLocalHost().getHostName())
+                    .put("language", "ballerina");
+        } catch (UnknownHostException e) {
+            console.println("Error while getting the host name");
```

Comment:
Is it possible to pass to the ballerina level and log the message instead of using console print?

Reply:
Addressed the review suggestions